frontend/src/pages/SubProjects/actions.js

```diff
@@ -101,9 +103,19 @@ export function setProjectHistoryOffset(offset) {
   };
 }
 
-export function fetchProjectHistory(projectId, offset, limit, showLoading = false) {
+export function fetchFirstProjectHistory(projectId, offset, limit, showLoading = false) {
   return {
-    type: FETCH_PROJECT_HISTORY,
+    type: FETCH_FIRST_PROJECT_HISTORY,
+    projectId,
+    offset,
+    limit,
+    showLoading
+  };
+}
+
+export function fetchNextProjectHistory(projectId, offset, limit, showLoading = false) {
+  return {
+    type: FETCH_NEXT_PROJECT_HISTORY,
     projectId,
     offset,
```

Comment:
Same as above. The redux action doesn't need to know how offset and limit works. What you ultimately want is the fetch the next `page`. Since we have a fixed `pageSize` we don't need to pass it. I would drop offset and limit completely and simply pass the `page` I want to fetch.